### PR TITLE
Skorohoda's first laboratory work

### PR DESCRIPTION
**Pull request by 0lezhka**

### CODE REVIEW

File: Labs/BestFriendsIntegrityUnitID 2/Dijkstra/src/main/java/com/kopylchak/dijkstra/controller/ConsoleController.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+                break;
+            case 3:
+                calculateTheBestWay();
+                break;
+            case 4:
+                performanceTest();
+                break;
+            case 5:
+                exit();
+        }
```
Comment (OstapMelnychuk) [inline Labs/BestFriendsIntegrityUnitID 2/Dijkstra/src/main/java/com/kopylchak/dijkstra/controller/ConsoleController.java:81]: No default case.

File: Labs/BestFriendsIntegrityUnitID 2/Dijkstra/src/main/java/com/kopylchak/dijkstra/graph/GraphImpl.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    public Set<Vertex> getVertexes() {
+        return graph;
+    }
+
+    @Override
+    public Vertex getVertex(final String name) {
+        return graph.stream()
+            .filter(v -> v.getName().equals(name))
+            .findFirst()
+            .orElse(null);
```
Comment (OstapMelnychuk) [inline Labs/BestFriendsIntegrityUnitID 2/Dijkstra/src/main/java/com/kopylchak/dijkstra/graph/GraphImpl.java:50]: Potentional Null Pointer Exception.

File: Labs/BestFriendsIntegrityUnitID 2/Dijkstra/src/main/java/com/kopylchak/dijkstra/graph/algorithms/ParallelDijkstra.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    @Override
+    protected Vertex findNextVertex(Vertex current, Map<Vertex, Long> marks, Set<Vertex> visited) {
+        return current.getNeighbors()
+            .keySet()
+            .parallelStream()
+            .filter(v -> !visited.contains(v))
+            .min(Comparator.comparing(marks::get))
+            .orElse(null);
```
Comment (OstapMelnychuk) [inline Labs/BestFriendsIntegrityUnitID 2/Dijkstra/src/main/java/com/kopylchak/dijkstra/graph/algorithms/ParallelDijkstra.java:39]: Possible NullPointerException.

File: Labs/BestFriendsIntegrityUnitID 2/Dijkstra/src/main/java/com/kopylchak/dijkstra/graph/io/ConsoleGraphPrinter.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    private void writeVertex(Vertex vertex, Graph graph) {
+        StringBuilder line = new StringBuilder(String.format(FROM_VERTEX_FORMAT, vertex.getName()));
+
+        for (Vertex v : vertex.getNeighbors().keySet()) {
+            line.append(String.format(TO_VERTEX_FORMAT, v.getName(), graph.getDistanceBetween(vertex, v)));
+        }
+
+        line.delete(line.length() - 2, line.length());
+
+        System.out.println((line.toString()));
```
Comment (OstapMelnychuk) [inline Labs/BestFriendsIntegrityUnitID 2/Dijkstra/src/main/java/com/kopylchak/dijkstra/graph/io/ConsoleGraphPrinter.java:45]: No need to call a "toString()" method

**Review by DmytroSavchuk — APPROVED** [review]

**Review by OstapMelnychuk — APPROVED** [review]

**Pull request merged**